### DISCUSSION TITLE
Allow custom params

### DISCUSSION THREAD

**Discussion by creazy231** [strapi/strapi#7081] [Legacy Topics]
I have an Endpoint to display some items: http://localhost:1337/inventory-items
but now I want to modify the response of `find` by using the `afterFind` hook but only when the url has the param `<URL>?dimensions=1`.
Doing so gives me the following error: `Your filters contain a field 'dimensions' that doesn't appear on your model definition nor it's relations`.

Is there a way to allow using custom params?

**Comment by sandiprb** [comment]
@creazy231

Why not make a custom controller to handle this?

**Reply by creazy231** [reply]
Yea, this could be an option but I thought its to overhead for what I wanted to do

**Reply by MattieBelt** [reply]
A simple custom controller is the way to go
